webapp/lib/controller.dart

```diff
@@ -90,6 +96,78 @@ void initUI() {
   );
 }
 
+Map<String, model.NeedsReplyData> getLatestDataForProjects(List<model.NeedsReplyData> updatedData) {
+  Map<String, model.NeedsReplyData> latestProjectData = {};
+  
+  for (var project in projectList) {
+    var data = updatedData.where((data) => data.project == project).toList();
+    
+    if (data.isEmpty) {
+      latestProjectData[project] = null;
+    } else {
+      latestProjectData[project] = data.last;
+    }
+  }
+  return latestProjectData;
+}
+
+bool isProjectStale(model.NeedsReplyData projectData) {
+  var now = new DateTime.now();
+  int lastUpdateTimeDiff =  now.difference(projectData.datetime).inHours;
+
+  if (lastUpdateTimeDiff >= 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
+  projectTimers[projectData.project].cancel();
+
+  if (stale) {
+    projectTimers[projectData.project] = null;
+  } else {
+    var timeToExecute =  projectData.datetime.add(Duration(hours: 1));
+    var now = new DateTime.now();
+    var duration = timeToExecute.difference(now);
+    var timer = new Timer(duration, () => {
+        if (projectData.project == view.contentView.projectSelectorView.selectedProject) {
+          view.contentView.stale = true
+        }
+      }
+    );
```

Comment:
Accidental syntax mistake from a leftover shorthand function? (the `=> {` bit translates as creating a map, which doesn't actually execute what's inside it properly!)
```suggestion
    var timer = new Timer(duration, () {
      if (projectData.project == view.contentView.projectSelectorView.selectedProject) {
        view.contentView.stale = true;
      }
    });
```